core/arch/arm/mm/core_mmu.c

```diff
@@ -988,6 +988,78 @@ static void set_pg_region(struct core_mmu_table_info *dir_info,
 	}
 }
 
+TEE_Result core_mmu_map_pages(vaddr_t start, paddr_t *pages, size_t num_pages,
+			      enum teecore_memtypes memtype)
+{
+	TEE_Result ret;
+	struct core_mmu_table_info tbl_info;
+	unsigned int idx;
+	uint32_t old_attr;
+	vaddr_t vaddr = start;
+	size_t i;
+
+	if (vaddr & SMALL_PAGE_MASK)
+		return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
I'm not very familiar with pager. If I got it right, pager holds list of owned pages in linked list. We will need to iterate over this list for every page being mapped?

We can check if VA corresponds to `MEM_AREA_RES_VASPACE` or `MEM_AREA_SHM_VASPACE`. Will be this enough?